lib/data_cycle/schedule_course_updates.rb

```diff
@@ -36,7 +37,11 @@ def run_update
 
   def enqueue_course_updates
     log_message "Ready to update #{Course.ready_for_update.count} courses"
-    Course.ready_for_update.each do |course|
+
+    courses_to_update = Course.ready_for_update
+    CheckCourseJobs.remove_orphan_locks(courses_to_update)
```

Comment:
What is the advantage of doing it with all courses in a batch, instead of one-by-one inside the block that also schedules each update?

Reply:
There are 2 reasons why I am doing it before the update block
- I did not like the idea of removing orphan locks, and executing jobs one by one as that would be kind of messy. I wanted to move forward with executing the jobs only when the previous "mistakes"(orphan locks) were solved.
- It is at times difficult for debugging.